tools/custom-checks/check-package-json-sort.ts

```diff
@@ -0,0 +1,122 @@
+import fs from "fs-extra";
+import path from "path";
+import { fileURLToPath } from "url";
+import { globby, Options as GlobbyOptions } from "globby";
+import { RuntimeError } from "run-time-error";
+import { isStdLibRecord } from "./is-std-lib-record";
+import { sortPackageJson } from "sort-package-json";
+
+export interface ICheckPackageJsonSort {
+  readonly argv: string[];
+  readonly env: NodeJS.ProcessEnv;
+  /**
+   * The version that will be used as the correct one when checking the sibling
+   * package dependency version declarations.
+   * If you omit this (optional) parameter then the root package.json file
+   * will be parsed to obtain its value at runtime.
+   */
+  readonly version?: string;
+}
+
+/**
+ * Sorts and checks that all the package.json files within the Cactus project
+ * Note: this only sorts and checks the package.json files that are within the
+ * `packages` folder for this proeject
+ *
+ * @returns An array with the first item being a boolean indicating
+ * 1) success (`true`) or 2) failure (`false`)
+ */
+export async function checkPackageJsonSort(
+  req: ICheckPackageJsonSort,
+): Promise<[boolean, string[]]> {
+  const TAG = "[tools/check-package-json-sort.ts]";
+  const __filename = fileURLToPath(import.meta.url);
+  const __dirname = path.dirname(__filename);
+  const SCRIPT_DIR = __dirname;
+  const PROJECT_DIR = path.join(SCRIPT_DIR, "../../");
+  console.log(`${TAG} SCRIPT_DIR=${SCRIPT_DIR}`);
+  console.log(`${TAG} PROJECT_DIR=${PROJECT_DIR}`);
+
+  if (!req) {
+    throw new RuntimeError(`req parameter cannot be falsy.`);
+  }
+  if (!req.argv) {
+    throw new RuntimeError(`req.argv cannot be falsy.`);
+  }
+  if (!req.env) {
+    throw new RuntimeError(`req.env cannot be falsy.`);
+  }
+
+  const globbyOpts: GlobbyOptions = {
+    cwd: PROJECT_DIR,
+    ignore: ["**/node_modules"],
+  };
+
+  const DEFAULT_GLOB = "**/cactus-*/package.json";
+
+  const pkgJsonPaths = await globby(DEFAULT_GLOB, globbyOpts);
+  sortPackageJson(pkgJsonPaths);
+
+  const lernaJsonPathAbs = path.join(PROJECT_DIR, "./lerna.json");
+  console.log(`${TAG} Reading root lerna.json at ${lernaJsonPathAbs}`);
+  const lernaJson = await fs.readJSON(lernaJsonPathAbs);
+  const correctVersion = req.version || lernaJson.version;
+  console.log(`${TAG} Correct Version: ${correctVersion}`);
+
+  const errors: string[] = [];
+
+  const checks = pkgJsonPaths.map(async (pathRel) => {
+    const filePathAbs = path.join(PROJECT_DIR, pathRel);
+    const pkgJson: unknown = await fs.readJSON(filePathAbs);
+    if (typeof pkgJson !== "object") {
+      errors.push(`ERROR: ${pathRel} package.json cannot be empty.`);
+      return;
+    }
+    if (!pkgJson) {
+      errors.push(`ERROR: ${pathRel} package.json cannot be empty.`);
+      return;
+    }
+    if (!isStdLibRecord(pkgJson)) {
+      return;
+    }
+
+    //   const { dependencies, devDependencies, version } = pkgJson;
```

Comment:
@Leeyoungone Please remove the commented code unless it is specifically needed to be kept around for some reason. (I'm guessing in this instance it is not but always happy to discuss!)

Reply:
I duplicated this file from another script in the`custom-checks` file. I commented this part of the code out because I thought I didn't need it, and i think I just forgot to delete it. Thank you :)